tests/foreign_object/tests.py

```diff
@@ -767,3 +768,12 @@ def test_pickling_foreignobject(self):
         foreign_object_restored = pickle.loads(pickle.dumps(foreign_object))
         self.assertIn("path_infos", foreign_object_restored.__dict__)
         self.assertIn("reverse_path_infos", foreign_object_restored.__dict__)
+
+
+class ForeignObjectConstraintTests(TestCase):
+    @skipUnlessDBFeature("supports_table_check_constraints")
+    def test_validate_constraints_with_foreign_key_id(self):
```

Comment:
Could you add a test case showing that `exclude={"patron_id"}` still avoids validating a failing constraint after this change? (I wouldn't worry about supporting `exclude={"patron"}`, which we've doc'd as [not working yet](https://docs.djangoproject.com/en/5.2/topics/composite-primary-key/#composite-primary-keys-in-model-validation), but I just want to ensure no regression here for the other case.

If you find your changes end up _fixing_ `exclude={"patron"}` e.g. through interrogating `local_related_fields`, then let's remove the doc'd limitation about specifying each local related field separately.